### PR TITLE
Incrementally rehash for incremental aggregations, use a specialized index builder for modifications.

### PR DESCRIPTION
**Pull request by cpwright**

### CODE REVIEW

**Review by cpwright — COMMENTED** [review]
Starting place holder.

**Review by cpwright — COMMENTED** [review]
Placeholder two on twosday.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/incopenagg/gen/IncrementalAggOpenHasherByteChar.java [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+            final char k1 = originalKeyArray1[sourceBucket];
+            final int hash = hash(k0, k1);
+            final int firstDestinationTableLocation = hashToTableLocation(hash);
+            int destinationTableLocation = firstDestinationTableLocation;
+            while (true) {
+                if (destState[destinationTableLocation] == EMPTY_OUTPUT_POSITION) {
+                    destKeyArray0[destinationTableLocation] = k0;
+                    destKeyArray1[destinationTableLocation] = k1;
+                    destState[destinationTableLocation] = originalStateArray[sourceBucket];
+                    if (sourceBucket != destinationTableLocation) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/incopenagg/gen/IncrementalAggOpenHasherByteChar.java:345]: This if statement needs to come out. We need to apply the hash.

**Review by cpwright — COMMENTED** [review]
Charles post-review placeholder.

**Review by rcaudy — APPROVED** [review]

**Pull request merged**